File: backend-ms1/src/main/java/ma/zs/easystock/service/facade/admin/stock/PaiementAchatAdminService.java
package ma.zs.easystock.service.facade.admin.stock;

import java.util.List;
import ma.zs.easystock.bean.core.stock.PaiementAchat;
import ma.zs.easystock.dao.criteria.core.stock.PaiementAchatCriteria;
import ma.zs.easystock.zynerator.service.IService;



public interface PaiementAchatAdminService extends  IService<PaiementAchat,PaiementAchatCriteria>  {

    List<PaiementAchat> findByAchatId(Long id);
    int deleteByAchatId(Long id);
    long countByAchatReference(String reference);



}

File: frontend/src/app/controller/service/admin/stock/PaiementAchatAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {environment} from 'src/environments/environment';

import {PaiementAchatDto} from 'src/app/controller/model/stock/PaiementAchat.model';
import {PaiementAchatCriteria} from 'src/app/controller/criteria/stock/PaiementAchatCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';
import {RoleAdminService} from './RoleAdmin.service';


@Injectable({
  providedIn: 'root'
})
export class PaiementAchatAdminService extends AbstractService<PaiementAchatDto, PaiementAchatCriteria> {
     constructor(private http: HttpClient, private roleService: RoleAdminService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/paiementAchat/');
    }
    getModelName(modelName: string): string {
        return this.modeleName = modelName;
    }

    public constrcutDto(): PaiementAchatDto {
        return new PaiementAchatDto();
    }

    public constrcutCriteria(): PaiementAchatCriteria {
        return new PaiementAchatCriteria();
    }
}
